tests/traits/InteractsWithTenancy.php

```diff
@@ -134,6 +135,18 @@ protected function setUpWebsites(bool $save = false, bool $connect = false)
         }
     }
 
+    protected function mockHttpRequest()
+    {
+        $this->app->singleton(Environment::class, function ($app) {
+            return new class($app) extends Environment {
+                public function runningInConsole()
```

Comment:
Interesting solution.  Nice job

Reply:
Thanks :-)

Reply:
If I'm not mistaken the app is only instantiated once per test class, your ioc override will affect all tests inside the class. I'm pretty sure that's not intended.

Reply:
I'll double check this, but I'm pretty sure it's not the case.

Reply:
Okay, so createApplication() is invoked once in setUp() and then flushed during tearDown(). So, the application is created once, but the ioc container is flushed for each test. To confirm this I commented out the second call to mockHttpRequest() in InstallationTest and that test failed.

Reply:
Actually, that's not quite right createApplication() is called in every setUp().

Reply:
Ah indeed, better safe than sorry 😊